### ISSUE TITLE
Models underestimating

### ISSUE THREAD

**Issue by Arnechos** [zalandoresearch/pytorch-ts#82]
Hello,
I've been trying to use DeepAR, TemporalFlow (Transformer aswell) and TimeGrad models for multivariate forecasting, however no matter what model I use the results are always negative, where training data is allways 0 or larger. Could you point me in the right direction? Increasing epochs makes results only worse.

**Comment by kashif** [comment]
right, so you mean with DeepVAR right? To fix the issue I would try DeepVAR with a D dimensional independent distribution first. 
Is it possible for you to share a snippet of the data so I can debug?

**Comment by Arnechos** [comment]
I've uploaded sample data in Google Drive saved via nump.save so you'll have to add in np.load argument allow_pickle=True.
https://drive.google.com/drive/folders/13Fu1ZkUN0UmFWMgXVi9fZ6scPB27rdmu?usp=sharing

As for the models, I'm facing the same issue with all the models, most recently with TimeGrand - to get a proper scale I had to multiply a number by 10^11

**Comment by kashif** [comment]
thanks! I'll have a look shortly!

**Comment by Arnechos** [comment]
Thanks! I forgot to add - data is at weekly aggregation

**Comment by Arnechos** [comment]
@kashif have you had some time to take a look at the data?

**Comment by kashif** [comment]
no not yet... let me look now...

**Comment by kashif** [comment]
@Arnechos so the first thing is that your test dataset has to be as large as the training (in time dim) and then prediction length more... so in your example it has to be (8, 32+72) dim

**Comment by Arnechos** [comment]
Oh, that solved all the issues I had. Thanks I lot! I haven't realized that's the correct way of handling train/test split in gluon

**Comment by kashif** [comment]
cool cool so with deepVAR:

```py
estimator = DeepVAREstimator(
    input_size = 23,
    num_cells= 16,
    prediction_length=32,
    context_length=32,
    distr_output=IndependentNormalOutput(dim=8),
    target_dim=8,
    freq='W',
    scaling=True,
    trainer=Trainer(device="cuda",
                    epochs=150,
                    learning_rate=1e-3,
                    num_batches_per_epoch=10,
                    batch_size=32,
                    num_workers=4)
)
```

I can get it to predict zeros for all dims...

**Comment by Arnechos** [comment]
We can close this issue. I've managed to get good results for my data. Thanks a lot for help and clarifications

**Comment by kashif** [comment]
awesome! thanks!
